src/field/field_factory.cxx

```diff
@@ -97,7 +97,13 @@ FieldFactory::~FieldFactory() {
 }
 
 const Field2D FieldFactory::create2D(const string &value, Options *opt, Mesh *m, CELL_LOC loc, BoutReal t) {
-  Field2D result = 0.;
+
+  if(m == NULL)
```

Comment:
It looks like this was copied from L113, but the original code is still in place -- delete L113-116?

Also, I know it's nitpicking, but this should be:

```cpp
if (m == nullptr) {
  m = fieldmesh;
}
```

Reply:
No, copied from the Field3D function - didn't notice that the check was not missing but just a few lines down.